symfit/core/objectives.py

```diff
@@ -124,6 +124,26 @@ def _invariant_kwargs(self):
         )
         return kwargs
 
+    def __eq__(self, other):
+        """
+        Objectives are considered equal if they are of the same type, have the
+        same model, and the same data.
+        """
+        if self.__class__ == other.__class__ and self.model == other.model:
```

Comment:
Why not allow subclassing?

Reply:
Not sure, should a subclass of an objective be the same as the original so long as it has the same data and model? I don't think so, right?

Reply:
That's the question. It's something to think about. Maybe just drop the class equality, and only compare attributes?

Reply:
I think the false positives one could get then are way worse. For example, even if I make a small modification like a least squares which returns the absolute magnitude in case of complex numbers, I would already not want that to compare as the same. So I think the class equality makes sense.

Reply:
Ok, add a comment with this reasoning, and leave it :)